Code/Tools/FBuild/FBuildCore/Graph/DependencyListNode.cpp

```diff
@@ -0,0 +1,243 @@
+        }
+
+        // Dump to txt file
+
+        FileStream dmp;
+        if ( dmp.Open( filename, FileStream::WRITE_ONLY ) == false )
+        {
+            return false;
+        }
+        else
+        {
+            dmp.WriteBuffer( buffer.Get(), buffer.GetLength() * sizeof(*buffer.Get()) );
+            dmp.Close();
+
+            return true;
+        }
+    }
+
+    static bool MakeListTxt( const Node * node, const char * filename, const Array< AString > & patterns )
+    {
+        DependencyListBuilder builder( patterns );
+
+        // Don't want to collect root node itself
+
+        builder.Collect( node->GetStaticDependencies() );
+        builder.Collect( node->GetDynamicDependencies() );
+
+        return builder.DumpTxtFile( filename );
+    }
+};
+
+// CONSTRUCTOR
+//------------------------------------------------------------------------------
+DependencyListNode::DependencyListNode()
+: FileNode( AString::GetEmpty(), Node::FLAG_NONE )
+{
+    m_Type = Node::DEPENDENCY_LIST_NODE;
+}
+
+// Initialize
+//------------------------------------------------------------------------------
+/*virtual*/ bool DependencyListNode::Initialize( NodeGraph & nodeGraph, const BFFToken * funcStartIter, const Function * function )
+{
+    // .PreBuildDependencies
+    if ( !InitializePreBuildDependencies( nodeGraph, funcStartIter, function, m_PreBuildDependencyNames ) )
+    {
+        return false; // InitializePreBuildDependencies will have emitted an error
+    }
+
+    // Get node for Source of dependency list
+    Node* srcNode = nodeGraph.FindNode( m_Source );
+    if ( srcNode != nullptr )
+    {
+        m_StaticDependencies.Append(Dependency( srcNode ));
+    }
+
+    return true;
+}
+
+// DoDynamicDependencies
+//------------------------------------------------------------------------------
+/*virtual*/ bool DependencyListNode::DoDynamicDependencies( NodeGraph & nodeGraph, bool forceClean )
+{
+    (void)forceClean; // dynamic deps are always re-added here, so this is meaningless
+
+    // clear dynamic deps from previous passes
+    m_DynamicDependencies.Clear();
+
+    // Get node for Source of dependency list
+    Node* srcNode = nodeGraph.FindNode( m_Source );
+    if ( srcNode == nullptr )
+    {
+        FLOG_ERROR( "failed to find '%s' for dependency list", m_Source.Get() );
+        return false;
+    }
+
+    m_DynamicDependencies.Append(Dependency( srcNode ));
+
+    return true;
+}
+
+// DESTRUCTOR
+//------------------------------------------------------------------------------
+DependencyListNode::~DependencyListNode() = default;
+
+// DoBuild
+//------------------------------------------------------------------------------
+/*virtual*/ Node::BuildResult DependencyListNode::DoBuild( Job * /*job*/ )
+{
+    FLOG_VERBOSE( "DependencyList: '%s' -> '%s'", m_Source.Get(), GetName().Get() );
+
+    // create the dependency list file
+    if ( DependencyListBuilder::MakeListTxt( this, m_Name.Get(), m_Patterns ) == false )
```

Comment:
Since you only have one source, and with the right metadata no alias traversal is needed, I don't think you need this helper class at all. You should be able to query the node for it's dependencies directly (they should already be unique)